pkg/openstack/nova.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	novav1 "github.com/openstack-k8s-operators/nova-operator/api/v1beta1"
+	corev1beta1 "github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// ReconcileNova -
+func ReconcileNova(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, helper *helper.Helper) (ctrl.Result, error) {
+	if !instance.Spec.Nova.Enabled {
+		return ctrl.Result{}, nil
+	}
+
+	nova := &novav1.Nova{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "nova",
+			Namespace: instance.Namespace,
+		},
+	}
+
+	helper.GetLogger().Info("Reconciling Nova", "Nova.Namespace", instance.Namespace, "Nova.Name", nova.Name)
+	op, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), nova, func() error {
+		// 1)
+		// Nova.Spec.APIDatabaseInstance and each NovaCell.CellDatabaseInstance
+		// aredefaulted to "openstack" in nova-operator and the MariaDB created
+		// by openstack-operator is also named "openstack". This works but
+		// in production we might want to have separate DB service instances
+		// per cell.
```

Comment:
Can we open a new issue for this point and link it here in this comment? That way we can discuss it and come up with a solution for this problem. If it remains here as a comment alone it might be overlooked for some time.

Reply:
I agree that we need to track these things but it is not clear how could I do it properly.
1. We have an Jira where I have ticket for this (OSP-20623). The ask was to create tracking tickets there. But that Jira is not public so I should not add that Jira ticket into this public repo.
2. I could create a github issue for this but then there would be a duplication between the internal Jira ticket and the external github action. Such duplication will lead to divergence which I dont like.

@dprince @stuggi : please advice what would be a good practice here?

Reply:
As discussed on slack, lets bring this up in our next weeks meeting.